src/core/distributions/phylogenetics/tree/birthdeath/StateDependentSpeciationExtinctionProcess.cpp

```diff
@@ -1919,61 +2000,13 @@ void StateDependentSpeciationExtinctionProcess::setValue(Tree *v, bool f )
     static_cast<TreeDiscreteCharacterData*>(this->value)->setCharacterData(tip_data);
    
     // simulate character history over the new tree
-    size_t num_nodes = value->getNumberOfNodes();
-    if (num_nodes > 2)
-    {
-        std::vector<std::string> character_histories(num_nodes);
-        drawStochasticCharacterMap(character_histories);
-    }
-    static_cast<TreeDiscreteCharacterData*>(this->value)->setTimeInStates(time_in_states);
-}
-
-
-std::vector<double> StateDependentSpeciationExtinctionProcess::calculateTotalSpeciationRatePerState( void ) 
-{
-    std::vector<double> total_rates = std::vector<double>(num_states, 0);
-    std::map<std::vector<unsigned>, double> eventMap;
-    std::vector<double> speciation_rates;
-    std::map<std::vector<unsigned>, double>::iterator it;
-    if ( use_cladogenetic_events == true )
-    {
-        // get cladogenesis event map (sparse speciation rate matrix)
-        eventMap = cladogenesis_matrix->getValue().getEventMap();
-        // iterate over each cladogenetic event possible
-        for (it = eventMap.begin(); it != eventMap.end(); it++)
-        {
-            const std::vector<unsigned>& states = it->first;
-            total_rates[states[0]] += it->second;
-        }
-    }
-    else
-    {
-        speciation_rates = lambda->getValue();
-        for (size_t i = 0; i < num_states; i++)
-        {
-            total_rates[i] += speciation_rates[i];    
-        }
-    }
-    return total_rates;
-}
-
-
-std::vector<double> StateDependentSpeciationExtinctionProcess::calculateTotalAnageneticRatePerState( void ) 
-{
-    std::vector<double> total_rates = std::vector<double>(num_states, 0);
-    const RateGenerator *rate_matrix = &getEventRateMatrix();
-    for (size_t i = 0; i < num_states; i++)
-    {
-        for (size_t j = 0; j < num_states; j++)
-        {
-            if (i != j)
-            {
-                total_rates[i] += rate_matrix->getRate(i, j, 0.0, getEventRate());
-            } 
-        }
-    }
-
-    return total_rates;
+//    size_t num_nodes = value->getNumberOfNodes();
```

Comment:
Should this be commented out -- maybe delete it along with the comment about simulating, if we are not simulating here?

Reply:
I'ld rather keep it if we need to revert ...